rclcpp/include/rclcpp/duration.hpp

```diff
@@ -29,17 +29,20 @@ class Duration
   RCLCPP_PUBLIC
   Duration(int32_t seconds, uint32_t nanoseconds);
 
+  // This constructor matches any numeric value - ints or floats
   RCLCPP_PUBLIC
-  explicit Duration(
-    rcl_duration_value_t nanoseconds);
+  explicit Duration(rcl_duration_value_t nanoseconds);
 
-  RCLCPP_PUBLIC
-  explicit Duration(
-    std::chrono::nanoseconds nanoseconds);
+  // This constructor matches any std::chrono value
+  template<class rep, class period>
+  explicit Duration(std::chrono::duration<rep, period> duration)
+  {
+    rcl_duration_.nanoseconds =
+      std::chrono::duration_cast<std::chrono::nanoseconds>(duration).count();
+  }
 
   RCLCPP_PUBLIC
-  Duration(
-    const builtin_interfaces::msg::Duration & duration_msg);
+  Duration(const builtin_interfaces::msg::Duration & duration_msg);
```

Comment:
According to CI, it looks like this constructor needs to be marked explicit.

Reply:
I didn't want to change that since it was unrelated to this PR. I think that change needs some discussion. IMHO, this constructor is better left implicit. With the implicit constructor, we can say:
```
rclcpp::Duration foo = mymsg.foo;
```
instead of 
```
rclcpp::Duration foo = rclcpp::Duration(mymsg.foo);
```

That seems to me to be a common enough use case that not being explicit would be convenient.

If you agree, I can mark this with // NOLINT

Reply:
This is a thing, it's called a conversion constructor:

https://en.cppreference.com/w/cpp/language/converting_constructor

And they _explicitly_ do not use the keyword `explicit` (ironically?), and in this case I think the linter is providing a hint that we should make sure we want this and in this case I believe we do. So I would +1 the use of `// NOLINT(runtime/explicit): this is a conversion constructor` here.

Reply:
Fine by me.  I was just confused by the fact that:

1.  There is no logical change to this line, so I'm not sure why the linter is picking it up now (maybe because of the line breaks?), and
1.  Why there is any change to this line in this PR.  Maybe putting it back to the old indentation will suffice?

Reply:
For sure, that's a valid question. If reverting the whitespace only change also quiets the linter that's fine too (though it doesn't make sense :p).

Reply:
Apparently the world is a strange place :-) I confirmed that reverting the spacing does indeed make the linter error go away.

I think that's a bit better than the //NOLINT at the end of the line, so I'll revert the space change.